packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -464,6 +474,7 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
   @override
   void dispose() {
     assert(!_transitionCompleter.isCompleted, 'Cannot dispose a $runtimeType twice.');
+    _performanceModeRequestHandle?.dispose();
```

Comment:
See earlier comment. Shouldn't we request the handle when the transition starts and dispose it when it ends? We don't need this for the lifetime of the route, which could be a loooong time?

Reply:
100% agreed, while looking at some of the timeline events the request was persisted until i went to a new page. Any ideas on what a good spot to dispose the handle would be?

Reply:
Can you put it on the animation status in _handleStatusChanged? Basically, you'd want to aquire a handle when the animation starts and dispose it when the animation is done.

Reply:
Done!